pkg/receive/hashring.go

```diff
@@ -151,8 +153,18 @@ func (c ketamaHashring) GetN(tenant string, ts *prompb.TimeSeries, n uint64) (st
 		i = 0
 	}
 
-	i = (i + n) % numSections
-	nodeIndex := c.sections[i].endpointIndex
+	ringSection := i % numSections
+	replicas := make(map[uint64]struct{})
```

Comment:
Allocating these maps can be expensive. I'll try to precompute the replica for each section when creating the ring.

Reply:
yea, benchmarks would be nice.

Reply:
The replica for each section should now be precalculated when the hashring is created, so these allocations have completely gone away.